numpy/core/tests/test_casting_unittests.py

```diff
@@ -695,6 +695,13 @@ def test_object_casts_NULL_None_equivalence(self, dtype):
             expected = arr_normal.astype(dtype)
         except TypeError:
             with pytest.raises(TypeError):
-                arr_NULLs.astype(dtype)
+                arr_NULLs.astype(dtype),
         else:
             assert_array_equal(expected, arr_NULLs.astype(dtype))
+
+    def test_float_to_bool(self):
+        # test case corresponding to gh-19514
+        # simple test for casting bool_ to float16 
+        res = np.array([0, 3, -7], dtype=np.int8).view(bool)
+        expected = [0, 1, 1]
+        assert_array_equal(res, expected)
```

Comment:
@yashasvimisra2798 this tes tis missing an `res.astype(np.float16)` to actually exercise the casting code, could you maybe make small PR to add that?

Reply:
Thanks, @seberg for pointing this out, I'll raise a PR for this.